enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/BoltInteraction.java

```diff
@@ -71,6 +71,7 @@
     public BoltInteraction( Neo4jWithSocket server ) throws IOException
     {
         this.server = server;
+        server.restartDatabase( settings -> {} );
```

Comment:
I took a different direction with the neo4j-admin user management work, which also showed leaking state between tests due to the temporary folder issue. I added code to delete both roles and auth to the Neo4jWithSocket itself. Not sure if we're fixing the exact same issue, but we could be.